src/game.cpp

```diff
@@ -738,7 +738,7 @@ void Game::playerMoveCreature(Player* player, Creature* movingCreature, const Po
 			}
 
 			Npc* movingNpc = movingCreature->getNpc();
-			if (movingNpc && !Spawns::isInZone(movingNpc->getMasterPos(), movingNpc->getMasterRadius(), toPos)) {
+			if (movingNpc && (!movingNpc->isPushable() || !Spawns::isInZone(movingNpc->getMasterPos(), movingNpc->getMasterRadius(), toPos))) {
```

Comment:
I think you may be right. I will analyze it again and check if this could be removed.